src/tiledb/cloud/utilities/common.py

```diff
@@ -0,0 +1,196 @@
+import configparser
+import logging
+import os
+import sys
+from typing import Any, Mapping
+
+import tiledb
+import tiledb.cloud
+
+
+def read_config(
+    path: str = "~/.aws/credentials",
+    section: str = "default",
+) -> Mapping[str, Any]:
+    """
+    Read config values from a file.
+
+    :param path: config file, defaults to "~/.aws/credentials"
+    :param section: section to read in the config file, defaults to "default"
+    :return: config dictionary
+    """
+    config = {}
+
+    cp = configparser.ConfigParser()
+    cp.read(os.path.expanduser(path))
+
+    keys = [
+        "aws_access_key_id",
+        "aws_secret_access_key",
+        "aws_session_token",
+        "aws_role_arn",
+        "aws_role_session_name",
+        "aws_external_id",
+    ]
+
+    for key in keys:
+        try:
+            config["vfs.s3." + key] = cp[section][key]
+        except KeyError:
+            pass
+
+    return config
+
+
+def set_context(config: Mapping[str, Any] = {}):
```

Comment:
Lets rename this to `set_aws_context`, or support more cloud vendors. Is there a reason this is setting os envs?

Reply:
We can rename this now and support more cloud vendors in the future. It's setting os envs for commands that access S3 directly, like AWS SDK CLI and bcftools.